### PR TITLE
Add Codesniffer to detect unused use-statements

### PR DESCRIPTION
**Pull request by nickvergessen**
(more to come)
### ToDo
- [x] Unused `use` statements:
  - [x] Fix instanceof `$a instanceof $b;`
  - [x] Fix closures `function($u, IU $e, $d, IUser $i){};`
  - [x] Fix type-comments `/* @var IQueryBuilder $qb */`
### Steps

```
cd build/codesniffer
composer install
./vendor/bin/phpcs -s -p --extensions=php --standard=nextcloud-unused.xml ../../apps/twofactor_backupcodes/
```

### CODE REVIEW

**Comment by nickvergessen** [comment]
Should be ready for review, is not yet executed, because a lot of things need fixing atm :see_no_evil:

File: build/codesniffer/phpbb/Sniffs/Namespaces/UnusedUseSniff.php [hunk 1]
```diff
@@ -0,0 +1,293 @@
+<?php
+/**
+ *
+ * This file is part of the phpBB Forum Software package.
+ *
+ * @copyright (c) phpBB Limited <https://www.phpbb.com>
+ * @license GNU General Public License, version 2 (GPL-2.0)
+ *
+ * For full copyright and license information, please see
+ * the docs/CREDITS.txt file.
```
Comment (ChristophWurst) [inline build/codesniffer/phpbb/Sniffs/Namespaces/UnusedUseSniff.php]: should we adjust this license header?

Comment (nickvergessen) [inline build/codesniffer/phpbb/Sniffs/Namespaces/UnusedUseSniff.php]: Right, I just c+p the file. But we are thinking about putting it into another project, then we could composer it. Until then I think we should just keep the copright+license, if that is okay @schiessle

Comment (ChristophWurst) [inline build/codesniffer/phpbb/Sniffs/Namespaces/UnusedUseSniff.php]: at least this line should be changed. Currently it's leading to a non-existing file

Comment (rullzer) [inline build/codesniffer/phpbb/Sniffs/Namespaces/UnusedUseSniff.php]: Ah yes we need the content of that file here then ;)

**Comment by rullzer** [comment]
Awesome stuff @nickvergessen 
Besides the licence thingy :+1:

File: build/codesniffer/phpbb/Sniffs/Namespaces/UnusedUseSniff.php (hunk 1 above)
Comment (nickvergessen) [inline build/codesniffer/phpbb/Sniffs/Namespaces/UnusedUseSniff.php]: Yeah, let's delay until it is split by phpBB into a decent package.

Comment (bantu) [inline build/codesniffer/phpbb/Sniffs/Namespaces/UnusedUseSniff.php]: I agree this should be split from phpBB. Cheers

**Comment by MorrisJobke** [comment]
@nickvergessen Could you have a look at this? Shouldn't be that hard to get this into shape

**Comment by nickvergessen** [comment]
I will continue this, once it was split into a separate composer package by upstream.

**Comment by MorrisJobke** [comment]
> I will continue this, once it was split into a separate composer package by upstream.

Any news here?

**Comment by nickvergessen** [comment]
Still on my todo, will close for now

**Pull request closed without merging**